engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -118,8 +211,131 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
+    val inputSymbols = actualInputs map {
+      case (name, value) => SymbolStoreEntry(name, value, None, input = true)
+    }
+
+    val callSymbols = for {
+      call <- workflowDescriptor.namespace.workflow.calls
+      (k, v) <- call.inputMappings
+    } yield SymbolStoreEntry(s"${call.fullyQualifiedName}.$k", v, None, input = true)
+
+    inputSymbols.toSet ++ callSymbols.toSet
+  }
+
+  /**
+    * Attempt to start all runnable calls and return updated state data.  This will create a new copy
+    * of the state data including new pending persists.
+    */
+  @tailrec
+  private def startRunnableCalls(data: WorkflowExecutionActorData): WorkflowExecutionActorData = {
+
+    val executionStore = data.executionStore
+
+    def upstreamEntries(entry: JobKey, prerequisiteScope: Scope): Seq[ExecutionStoreEntry] = {
+      prerequisiteScope.closestCommonAncestor(entry.scope) match {
+        /**
+          * If this entry refers to a Scope which has a common ancestor with prerequisiteScope
+          * and that common ancestor is a Scatter block, then find the shard with the same index
+          * as 'entry'.  In other words, if you're in the same scatter block as your pre-requisite
+          * scope, then depend on the shard (with same index).
+          *
+          * NOTE: this algorithm was designed for ONE-LEVEL of scattering and probably does not
+          * work as-is for nested scatter blocks
+          */
+        case Some(ancestor: Scatter) =>
+          executionStore filter { case (k, _) => k.scope == prerequisiteScope && k.index == entry.index } toSeq
+
+        /**
+          * Otherwise, simply refer to the entry the collector entry.  This means that 'entry' depends
+          * on every shard of the pre-requisite scope to finish.
+          */
+        case _ =>
+          executionStore filter { case (k, _) => k.scope == prerequisiteScope && k.index.isEmpty } toSeq
+      }
+    }
+
+    def isDone(e: JobKey, status: ExecutionStatus): Boolean =
+      executionStore exists { case (k, s) => k.scope == e.scope && k.index == e.index && s == ExecutionStatus.Done }
+
+    def arePrerequisitesDone(key: JobKey): Boolean = {
+      val upstream = key.scope.prerequisiteScopes.map(s => upstreamEntries(key, s))
+      val dependencies = upstream.flatten
+      // TODO: Check downstream for Scatter calls (?)
+      val dependenciesResolved = executionStore.filter(dependencies) forall (x => isDone(x._1, x._2))
+
+      /**
+        * We need to make sure that all prerequisiteScopes have been resolved to some entry before going forward.
+        * If a scope cannot be resolved it may be because it is in a scatter that has not been populated yet,
+        * therefore there is no entry in the executionStore for this scope.
+        * If that's the case this prerequisiteScope has not been run yet, hence the (upstream forall {_.nonEmpty})
+        */
+      (upstream forall {
+        _.nonEmpty
+      }) && dependenciesResolved
```

Comment:
Just thinking out loud: I don't know the origin of all this but it seems kinda backwards. It all seems to come about because val dependenciesResolved = executionStore.filter(dependencies) forall (x => isDone(x._1, x._2))` is true even if the executionStore doesn't contain a dependency.

But anyway, maybe for this non-scatter ticket can we just remove this and whoever follows with the scatter work can re-introduce it if necessary?